xrpl4j-integration-tests/src/test/java/org/xrpl/xrpl4j/tests/NfTokenIT.java

```diff
@@ -0,0 +1,475 @@
+import org.xrpl.xrpl4j.model.flags.Flags;
+import org.xrpl.xrpl4j.model.ledger.NfTokenOfferObject;
+import org.xrpl.xrpl4j.model.transactions.NfTokenAcceptOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenBurn;
+import org.xrpl.xrpl4j.model.transactions.NfTokenCancelOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenCreateOffer;
+import org.xrpl.xrpl4j.model.transactions.NfTokenId;
+import org.xrpl.xrpl4j.model.transactions.NfTokenMint;
+import org.xrpl.xrpl4j.model.transactions.NfTokenUri;
+import org.xrpl.xrpl4j.model.transactions.TransactionResultCodes;
+import org.xrpl.xrpl4j.model.transactions.XrpCurrencyAmount;
+import org.xrpl.xrpl4j.tests.environment.CustomEnvironment;
+import org.xrpl.xrpl4j.tests.environment.XrplEnvironment;
+import org.xrpl.xrpl4j.wallet.Wallet;
+
+public class NfTokenIT extends AbstractIT {
+
+  private static final XrplEnvironment nftDevnetEnvironment = new CustomEnvironment(
+    HttpUrl.parse("http://xls20-sandbox.rippletest.net:51234"),
+    HttpUrl.parse("https://faucet-nft.ripple.com")
+  );
+  private final XrplClient xrplClient = xrplClient();
+
+  @Test
+  void mint() throws JsonRpcClientErrorException {
+    Wallet wallet = createRandomAccount();
+
+    AccountInfoResult accountInfoResult = this.scanForResult(
+      () -> this.getValidatedAccountInfo(wallet.classicAddress())
+    );
+
+    NfTokenUri uri = NfTokenUri.ofPlainText("ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf4dfuylqabf3oclgtqy55fbzdi");
+
+    //Nft mint transaction
+    NfTokenMint nfTokenMint = NfTokenMint.builder()
+      .tokenTaxon(UnsignedLong.ONE)
+      .account(wallet.classicAddress())
+      .fee(XrpCurrencyAmount.ofDrops(50))
+      .signingPublicKey(wallet.publicKey())
+      .sequence(accountInfoResult.accountData().sequence())
+      .uri(uri)
+      .build();
+
+    SubmitResult<NfTokenMint> mintSubmitResult = xrplClient.submit(wallet, nfTokenMint);
+    assertThat(mintSubmitResult.result()).isEqualTo(TransactionResultCodes.TES_SUCCESS);
+    assertThat(mintSubmitResult.transactionResult().transaction().hash()).isNotEmpty().get()
+      .isEqualTo(mintSubmitResult.transactionResult().hash());
+
+    this.scanForResult(
+      () -> {
+        try {
+          return xrplClient.accountNfts(wallet.classicAddress());
+        } catch (JsonRpcClientErrorException e) {
+          throw new RuntimeException(e);
+        }
+      },
+      result -> result.accountNfts().stream()
+        .anyMatch(nft -> nft.uri().get().equals(uri))
+    );
+    logger.info("NFT was minted successfully.");
+  }
+
+  @Test
+  void mintAndBurn() throws JsonRpcClientErrorException {
+    Wallet wallet = createRandomAccount();
+
+    AccountInfoResult accountInfoResult = this.scanForResult(
+      () -> this.getValidatedAccountInfo(wallet.classicAddress())
+    );
+
+    // nft mint
+    NfTokenUri uri = NfTokenUri.ofPlainText("ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf4dfuylqabf3oclgtqy55fbzdi");
```

Comment:
You can't use the same URI for all of the NFTs you are minting in these ITs and reliably use the check below. If this test runs after `mint()` for example, that NFT will already exist, and you won't actually be checking that the NFT you are minting in this test got minted.

Reply:
In each of these tests, we have different accounts generating them. So verifying shouldn't be a problem.

Reply:
Ah good call. Nevermind then